modules/address/components/AddressIconBox.tsx

```diff
@@ -29,15 +29,27 @@ export default function AddressIconBox({
   width = 41,
   limitTextLength = 0
 }: PropTypes): React.ReactElement {
+  // Early return if address is not provided to prevent toLowerCase errors
+  if (!address) {
+    return (
+      <Flex>
+        <Flex sx={{ minWidth: width, mr: 2, alignItems: 'center' }}>
+          <Text sx={{ fontSize: 1, color: 'textMuted' }}>Invalid address</Text>
+        </Flex>
+      </Flex>
+    );
+  }
+
```

Comment:
Given that there are 3 hook calls below this early return (`useNetwork`, `useAccount` and `useSingleDelegateInfo`), I wonder if early returning could cause the react hook count mismatch issue.
Perhaps we should move this below those hook calls, but before the `isOwner` declaration?